tests/data_model/test_schemaorg_normalizer.py

```diff
@@ -0,0 +1,355 @@
+
+    def test_toJsonLd(self):
+        given = [
+            {
+                "@id": "http://example.com/123",
+                "http://schema.org/name": [{"@value": "test"}],
+                "http://schema.org/additionalName": [{
+                    "@id": "http://example.com/124"
+                }]
+            },
+            {
+                "@id": "http://example.com/124",
+                "http://schema.org/firstName": [{"@value": "Lola"}],
+                "http://schema.org/lastName": [{"@value": "Showgirl"}]
+            }
+        ]
+
+        result = SchemaOrgGraph(given).toJsonLd()
+
+        assert result == given, "Should return flattened expanded json-ld format"
+
+
+class TestSchemaOrgNormalizer:
+    def test_normalizeFromExtruct(self):
+        given = {
+            'some_key': [
+                {
+                    "@context": "http://schema.org",
+                    "@id": "http://example.com/123",
+                    "name": [{"@value": "test"}],
+                    "additionalName": [{
+                        "@id": "http://example.com/124"
+                    }]
+                },
+                {
+                    "@context": "http://schema.org",
+                    "@id": "http://example.com/124",
+                    "firstName": [{"@value": "Lola"}],
+                    "lastName": [{"@value": "Showgirl"}]
+                }
+            ]
+        }
+
+        expected = [
+            {
+                "http://schema.org/name": ["test"],
+                "http://schema.org/additionalName": [{
+                    "http://schema.org/firstName": ["Lola"],
+                    "http://schema.org/lastName": ["Showgirl"]
+                }]
+            },
+            {
+                "http://schema.org/firstName": ["Lola"],
+                "http://schema.org/lastName": ["Showgirl"]
+            }
+        ]
+
+        result = SchemaOrgNormalizer.normalizeFromExtruct(given)
+
+        assert result == expected, "Should normalize nested compacted data"
+
+    def test_normalizeFromExtruct_multiple(self):
+        given = {
+            'some_key': [
+                {
+                    "@context": "http://schema.org",
+                    "@id": "http://example.com/123",
+                    "name": [{"@value": "test"}],
+                    "additionalName": [{
+                        "@id": "http://example.com/124"
+                    }]
+                },
+                {
+                    "@context": "http://schema.org",
+                    "@id": "http://example.com/124",
+                    "firstName": [{"@value": "Lola"}],
+                    "lastName": [{"@value": "Showgirl"}]
+                }
+            ],
+            'some_other_key': [
+                {
+                    "@type": "http://schema.org/Thing",
+                    "http://schema.org/name": "Lovely Test"
+                }
+            ]
+        }
+
+        expected = [
```

Comment:
Maybe I'm missing something obvious but this expected result doesn't follow the structure of SS3: https://github.com/wmde/reference-island#ss3-normalize-data-from-various-scraped-raw-formats

Reply:
As we discussed on mattermost, it actually doesn't make sense as a standalone that reads from disk, as we need to feed the scraped result directly to it. That's why I asked you a couple of weeks ago how you would like to have the data, and we decided on the format shown here.

This is a valid point though, as I need to also update the docs to reflect that.

Reply:
Thanks. I'm not saying this should change, I'm saying they are not consistent which might be confusing (I didn't know which direction is correct), thank you for fixing the README.

Reply:
No problem, sorry if I seemed defensive, I was just worried about a misunderstaning.